irony-cdb.el

```diff
@@ -114,6 +128,24 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--locate-dominating-file-with-dirs (file name subdirectories)
+  "Convenience wrapper around `locate-dominating-file'
+
+Looks up the directory hierarchy from FILE for to locate any directory
+in `subdirectories` which contains NAME. Stops at the first parent
+directory where the test succeeds and returns the full path to file
+found."
+  (let ((check-subdirs
```

Comment:
1. No. the behavior is to find the matching subdirectory in the nearest parent that matches.
if I `(locate-dominating dir ".")` and locate `../../<>.json` finds one 2 levels up, It won't check for the `one ../build/<>.json`. Doing it this way I find the closest one, and potentially check less directories.

2. `locate-dominating-file` returns the directory, not the path. So I have to reuse this check 
to figure out in which subdir the file was found. Since I reuse the logic (and it's not a one liner), this is simply DRY.